cmake/external/grpc.cmake

```diff
@@ -22,9 +22,9 @@ ExternalProject_Add(
   grpc-download
 
   DOWNLOAD_DIR ${FIREBASE_DOWNLOAD_DIR}
-  DOWNLOAD_NAME grpc-1.8.3.tar.gz
-  URL https://github.com/grpc/grpc/archive/v1.8.3.tar.gz
-  URL_HASH SHA256=c14bceddc6475a09927a815811a8161cdfa7acb445262835da6bc24da9842c92
+  DOWNLOAD_NAME grpc-1.12.0.tar.gz
```

Comment:
@wilhuff PTAL at this change. gRPC now prefers `#include "grpcpp/foo.h"` over the deprecated `#include "grpc++/foo.h"`, but the version we had there previously only supports the older style. This made Travis CMake build fail. I bumped the version to the latest stable release.

Reply:
@wilhuff Friendly ping.

Reply:
Sorry I didn't see this in my list of assigned PRs and didn't know it was waiting on me :-(.

In general I'm in favor of upgrading. Is there anything special about 1.12 instead of something nearer to head?

Objective-C clients that install today are using 1.15. That version includes grpc/grpc#16358 which may require additional build work from us (though I'm not sure). Maybe 1.14.2 might be a reasonable next upgrade?